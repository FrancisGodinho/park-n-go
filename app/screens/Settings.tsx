import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import React from "react";
import { signOut } from "@firebase/auth";
import { auth } from "../utils/Firebase";

type Props = {};

const Settings = (props: Props) => {
  const handleSignOut = () => {
    signOut(auth).catch((err) => console.error(err));
  };

  handleSignOut();

  return (
    <View>
      <Text>Settings</Text>
      <TouchableOpacity onPress={handleSignOut}></TouchableOpacity>
    </View>
  );
};

export default Settings;

const styles = StyleSheet.create({});
