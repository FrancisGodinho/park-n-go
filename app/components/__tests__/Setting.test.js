import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react-native";
import Settings from "../../screens/Settings";

beforeAll((done) => {
  done();
});

afterAll((done) => {
  done();
});

let findTextElement = function (tree, element) {
  console.warn(tree);
  return true;
};

describe("<Settings />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Settings text element exists", () => {
    const { getByText } = render(<Settings />);
    const text = getByText("Settings");
    expect(text).toBeTruthy();
  });
});
