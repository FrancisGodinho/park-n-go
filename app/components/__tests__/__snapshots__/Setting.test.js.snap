// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<View>
  <Text>
    Settings
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</View>
`;
